xlml/apis/task.py

```diff
@@ -243,11 +262,15 @@ class TpuXpkTask(BaseTask):
     task_test_config: Test configs to run on this TPU.
     task_gcp_config: Runtime TPU creation parameters.
     task_metric_config: Metric configs to process metrics.
+    prepend_tasks: Custom tasks to run before running main test job
+    append_tasks: Custom tasks to run after running main test job
   """
 
   task_test_config: test_config.TestConfig[test_config.Tpu]
   task_gcp_config: gcp_config.GCPConfig
   task_metric_config: Optional[metric_config.MetricConfig] = None
+  prepend_tasks: DAGNode = None
+  append_tasks: DAGNode = None
```

Comment:
I'd like to avoid adding these if possible. Using too many one-offs on the infra side can make things harder to maintain long-term. The use case here (i.e. creating random bucket path) to me seems like it should be common to all tests.

cc @RissyRan since this is more infra-side, keen to hear your thoughts.

Reply:
I agree with you, Jon. I also suggest to have two separate run tasks, like `run()` or `run_with_startup_script()`.

Reply:
Sounds good, I created a new run task as suggested.